tests/sdk/nodejs/dry-run/step2/index.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016-2021, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as k8s from "@pulumi/kubernetes";
+import * as random from "@pulumi/random";
+import * as pulumi from "@pulumi/pulumi";
+
+// TODO(levi): Use auto-naming for namespace once https://github.com/pulumi/pulumi-kubernetes/issues/1632 is fixed.
+const suffix = new random.RandomString("suffix", {length: 7, special: false, upper: false}).result;
+const ns = new k8s.core.v1.Namespace("test", {metadata: {name: pulumi.interpolate`test-${suffix}`}});
+const provider = new k8s.Provider("k8s", {enableDryRun: true}); // Use the default namespace.
```

Comment:
Consider adding a comment explaining what we are testing here, it's bit hard to spot the difference between step1 and step2 and understand why we are doing this

Reply:
I'll add a comment, but the idea is that the Provider in step1 specifies a namespace, and then removes that option in step2, which replaces the Deployment in the default namespace. This is equivalent to changing `.metadata.namespace` on the Deployment.